test/View/RouteNotFoundStrategyTest.php

```diff
@@ -288,35 +291,33 @@ public function testNotFoundTemplateIsMutable()
     public function testAttachesListenersAtExpectedPriorities()
     {
         $events = new EventManager();
-        $events->attachAggregate($this->strategy);
+        $this->strategy->attach($events);
 
         foreach ([MvcEvent::EVENT_DISPATCH => -90, MvcEvent::EVENT_DISPATCH_ERROR => 1] as $event => $expectedPriority) {
-            $listeners        = $events->getListeners($event);
-            $expectedCallback = [$this->strategy, 'prepareNotFoundViewModel'];
+            $listeners        = $this->getListenersForEvent($event, $events, true);
+            $expectedListener = [$this->strategy, 'prepareNotFoundViewModel'];
             $found            = false;
-            foreach ($listeners as $listener) {
-                $callback = $listener->getCallback();
-                if ($callback === $expectedCallback) {
-                    if ($listener->getMetadatum('priority') == $expectedPriority) {
-                        $found = true;
-                        break;
-                    }
+            foreach ($listeners as $priority => $listener) {
```

Comment:
Working on that now. It's actually an argument for a trait, as then I can do assertions within a method. I'm creating `assertListenerAtPriority(callable $expectedListener, $expectedPriority, $event, EventManager $events, $message = '')` right now; greatly reducing a ton of this boilerplate, and making the test cases easier to understand.